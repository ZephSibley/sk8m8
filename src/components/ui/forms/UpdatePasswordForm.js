import React from 'react';
import Button from '@material-ui/core/Button';

import formStyles from '../../../styles/forms';
import ValidInput from '../ValidInput';

const UpdatePasswordForm = () => {
    const formClasses = formStyles();
    
    return (
        <form className={formClasses.form}>
            <ValidInput
                label='password'
                validationrules={{
                    minLength: 6,
                }}
            />
            <ValidInput
                label='confirm password'
                validationrules={{
                    equalTo: 'password', //TODO
                }}
            />
            <input
                style={{ display: 'none' }}
                id="button-submit"
                multiple
                type="submit"
                // TODO add submit target
            />
            <label htmlFor="button-submit">
                <Button
                    variant="contained"
                    color="primary"
                    component="span"
                    className={formClasses.submit}
                >
                    Update Password
                </Button>
            </label>
        </form>
    )
}

export default UpdatePasswordForm;