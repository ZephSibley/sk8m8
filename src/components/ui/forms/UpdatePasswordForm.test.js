import React from 'react';
import renderer from 'react-test-renderer';
import UpdatePasswordForm from './UpdatePasswordForm';

test('Basic Rendering', () => {
    const component = renderer.create(
        <UpdatePasswordForm />
    )
    let tree = component.toJSON
    expect(tree).toMatchSnapshot();
})