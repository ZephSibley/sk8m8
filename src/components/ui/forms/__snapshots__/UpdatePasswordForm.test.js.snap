// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<form
  className="makeStyles-form-1"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      password
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
        value=""
      />
    </div>
    
  </div>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      confirm password
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        name="confirm_password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
        value=""
      />
    </div>
    
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root makeStyles-submit-5 MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
    disabled={true}
    margin="normal"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Update Password
    </span>
  </button>
</form>
`;
